library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/datasketches/cardinality/serialisation/json/HllSketchJsonDeserialiser.java

```diff
@@ -17,52 +17,27 @@
 package uk.gov.gchq.gaffer.sketches.datasketches.cardinality.serialisation.json;
 
 import com.fasterxml.jackson.core.JsonParser;
-import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.TreeNode;
 import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.JsonDeserializer;
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.node.ArrayNode;
-import com.fasterxml.jackson.databind.node.IntNode;
-import com.fasterxml.jackson.databind.node.TextNode;
-import com.yahoo.sketches.hll.HllSketch;
+import org.apache.datasketches.hll.HllSketch;
 
-import java.io.IOException;
-
-import static java.util.Objects.isNull;
-import static java.util.Objects.nonNull;
-import static uk.gov.gchq.gaffer.sketches.datasketches.cardinality.serialisation.json.HllSketchJsonConstants.DEFAULT_LOG_K;
-import static uk.gov.gchq.gaffer.sketches.datasketches.cardinality.serialisation.json.HllSketchJsonConstants.VALUES;
+import uk.gov.gchq.gaffer.exception.SerialisationException;
+import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
 
 /**
- * A {@code HyperLogLogPlusJsonSerialiser} deserialises {@link HllSketch} objects.
+ * A {@code HllSketchJsonDeserialiser} deserialises {@link HllSketch} objects.
  */
 public class HllSketchJsonDeserialiser extends JsonDeserializer<HllSketch> {
 
     @Override
-    public HllSketch deserialize(final JsonParser jsonParser, final DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
-        final TreeNode treeNode = jsonParser.getCodec().readTree(jsonParser);
-
-        final HllSketch hllp;
-
-        final TextNode jsonNodes = (TextNode) treeNode.get(HllSketchJsonConstants.BYTES);
-        if (isNull(jsonNodes)) {
-            final IntNode kNode = (IntNode) treeNode.get(HllSketchJsonConstants.LOG_K);
-            final int k = nonNull(kNode) ? kNode.asInt(DEFAULT_LOG_K) : DEFAULT_LOG_K;
-            hllp = new HllSketch(k);
-        } else {
-            hllp = HllSketch.heapify(jsonNodes.binaryValue());
+    public HllSketch deserialize(final JsonParser jsonParser, final DeserializationContext deserializationContext) throws SerialisationException {
+        try {
+            final TreeNode treeNode = jsonParser.getCodec().readTree(jsonParser);
+            final HllSketchWithValues hllSketchWithValues = JSONSerialiser.deserialise(treeNode.toString(), HllSketchWithValues.class);
+            return hllSketchWithValues.getHllSketch();
+        } catch (final Exception e) {
+            throw new SerialisationException("Error deserialising JSON object: " + e.getMessage(), e);
```

Comment:
What's happening here? The new code seems to be much simpler than before, where's the previous logic gone?

Reply:
Like the HyperLogLogPlusJsonDeserialiser + HyperLogLogPlusWithOffers change, this class has been changed so the logic now lives within HllSketchWithValues